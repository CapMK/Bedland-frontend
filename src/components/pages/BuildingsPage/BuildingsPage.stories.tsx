import type { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import BuildingsPage from './BuildingsPage';

const meta = {
  title: 'Pages/BuildingsPage',
  component: BuildingsPage,
  tags: ['autodocs'],
  parameters: {
    backgrounds: { default: 'light' },
  },
  decorators: [
    (Story) => (
      <BrowserRouter>
        <ThemeProvider>
          <GlobalStyle />
          {Story()}
        </ThemeProvider>
      </BrowserRouter>
    ),
  ],
} satisfies Meta<typeof BuildingsPage>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {};
