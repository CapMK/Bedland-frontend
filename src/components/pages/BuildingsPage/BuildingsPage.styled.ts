import styled from 'styled-components';

export const SortSearchContainer = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-between;
  margin: 1.5rem 0 1rem;
`;

export const BuildingsPageContent = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;
