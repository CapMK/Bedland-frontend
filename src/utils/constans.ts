export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
export const LOGIN_INPUT_WIDTH = 19.875;

// TODO
// BASE_URL points to json server
// Replace with actual API url when BE is ready
export const BASE_URL = 'http://localhost:3001';
