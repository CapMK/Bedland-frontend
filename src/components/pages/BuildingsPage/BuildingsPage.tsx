import { useState, useEffect } from 'react';
// eslint-disable-next-line import/no-named-default
import { default as BuildingsList } from 'components/organisms/DisplayList/DisplayList';
import Page from 'components/templates/Page/Page';
import mockBuildingsList from 'utils/mock/mockBuildingsList';
import SearchBar from 'components/molecules/SearchBar/SearchBar';
import SortButton from 'components/molecules/SortButton/SortButton';
import { sortType } from 'components/molecules/SortButton/SortButton.types';
import { DisplayListType } from 'components/organisms/DisplayList/DisplayList.types';
import { sortList } from 'components/molecules/SortButton/SortButton.utils';
import * as Styled from './BuildingsPage.styled';

const BuildingsPage = () => {
  const [displayList, setDisplayList] = useState<DisplayListType>([]);

  useEffect(() => {
    setDisplayList([...sortList(mockBuildingsList, 'name')] as DisplayListType);
  }, []);

  const changeSortOption = (sortOption: sortType): void => {
    setDisplayList([...sortList(displayList, sortOption)] as DisplayListType);
  };

  return (
    <Page title='Buildings'>
      <Styled.BuildingsPageContent>
        <Styled.SortSearchContainer>
          <SortButton text='Sort Buildings' changeSortOption={changeSortOption} />
          <SearchBar placeholder='Search buildings...' />
        </Styled.SortSearchContainer>
        <BuildingsList displayList={displayList} />
      </Styled.BuildingsPageContent>
    </Page>
  );
};
export default BuildingsPage;
