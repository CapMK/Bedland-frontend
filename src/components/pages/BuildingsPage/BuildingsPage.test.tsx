import { cleanup, screen, within, fireEvent } from '@testing-library/react';
import { renderWithProviders } from 'utils/test.utils';
import { loggedInManagerUser } from 'utils/mockUser';
import BuildingsPage from './BuildingsPage';

describe('BuildingsPage', () => {
  afterEach(cleanup);
  it('renders correctly', () => {
    const page = renderWithProviders(<BuildingsPage />, {
      preloadedState: {
        user: loggedInManagerUser,
      },
    });
    expect(page).toMatchSnapshot();
  });

  it('renders with buildings sorted by name', () => {
    renderWithProviders(<BuildingsPage />, {
      preloadedState: {
        user: loggedInManagerUser,
      },
    });
    const buildingsList = screen.getByRole('list');
    const { getAllByRole } = within(buildingsList);
    const items = getAllByRole('listitem');
    expect(items[0]).toHaveTextContent('Opole, Poland');
  });

  it('buildings get sorted by city', () => {
    renderWithProviders(<BuildingsPage />, {
      preloadedState: {
        user: loggedInManagerUser,
      },
    });
    const sortButton = screen.getByRole('button', {
      name: /Sort Buildings/i,
    });
    fireEvent.click(sortButton);
    const sortOption = screen.getByText(/city/i);
    fireEvent.click(sortOption);
    const buildingsList = screen.getByRole('list');
    const { getAllByRole } = within(buildingsList);
    const items = getAllByRole('listitem');
    expect(items[0]).toHaveTextContent('Kraków, Poland');
  });

  it('buildings get sorted by date added', () => {
    renderWithProviders(<BuildingsPage />, {
      preloadedState: {
        user: loggedInManagerUser,
      },
    });
    const sortButton = screen.getByRole('button', {
      name: /Sort Buildings/i,
    });
    fireEvent.click(sortButton);
    const sortOption = screen.getByText(/date added/i);
    fireEvent.click(sortOption);
    const buildingsList = screen.getByRole('list');
    const { getAllByRole } = within(buildingsList);
    const items = getAllByRole('listitem');
    expect(items[0]).toHaveTextContent('Kraków, Poland');
  });
});
