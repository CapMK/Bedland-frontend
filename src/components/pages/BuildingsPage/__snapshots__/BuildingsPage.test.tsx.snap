// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BuildingsPage > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c15 {
  margin-top: 1rem;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #fff;
  border-radius: 0.5rem;
  color: #1B1A1A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c6:hover {
  background-color: #fff;
  border: 1px solid #fff;
  color: #6573FF;
}

.c16 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4553DD;
  border: 1px solid #4553DD;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c16:hover {
  background-color: #6573FF;
  border: 1px solid #6573FF;
}

.c12 {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0px 5px 8px #D9D9D920;
  color: #4B4B4B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c13 {
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
}

.c14 {
  padding: 0.5rem 1.031rem 0.938rem;
}

.c11 {
  display: grid;
  gap: 2rem;
  grid-template-columns: repeat(auto-fit,minmax(20rem,1fr));
}

.c0 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1rem;
  position: relative;
  width: 100%;
  z-index: 1;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #fff;
  border-bottom: 1px solid transparent;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.375rem 0rem 0.375rem 0.875rem;
  width: 17.5rem;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  gap: 0.5625rem;
  height: 1.75rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c9 {
  font-family: Source Sans Pro,sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  width: 100%;
}

.c10 {
  position: relative;
}

.c5 {
  position: relative;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 1.5rem 0 1rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <h2
            style="font-family: Poppins, sans-serif; font-weight: 700; font-size: 1.5rem; line-height: 1.4; color: rgb(27, 26, 26);"
          >
            Buildings
          </h2>
        </div>
        <div
          class="c2"
        >
          <div
            class="c3"
          >
            <div
              class="c4"
            >
              <div
                class="c5"
              >
                <button
                  class="c6"
                >
                  <svg
                    data-testid="arrowDown-icon"
                    fill="none"
                    height="12"
                    viewBox="0 0 12 7"
                    width="12"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <mask
                      id="mask-arrowDown"
                    />
                    <path
                      d="M10.6667 1.33325L6.08333 5.49992L1.5 1.33325"
                      stroke="#1B1A1A"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1.67"
                    />
                  </svg>
                  Sort Buildings
                </button>
              </div>
              <div>
                <div
                  class="c7"
                >
                  <div
                    class="c8"
                  >
                    <svg
                      data-testid="search-icon"
                      fill="none"
                      height="20"
                      viewBox="0 0 16 16"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <mask
                        id="mask-search"
                      />
                      <path
                        d="M14.6666 14.6666L11.2916 11.2916M11.2916 11.2916C11.8333 10.7499 12.263 10.1068 12.5562 9.39907C12.8494 8.6913 13.0003 7.93272 13.0003 7.16664C13.0003 6.40055 12.8494 5.64197 12.5562 4.9342C12.263 4.22644 11.8333 3.58334 11.2916 3.04164C10.7499 2.49994 10.1068 2.07023 9.39907 1.77707C8.6913 1.4839 7.93272 1.33301 7.16664 1.33301C6.40055 1.33301 5.64197 1.4839 4.9342 1.77707C4.22644 2.07023 3.58334 2.49994 3.04164 3.04164C1.94762 4.13566 1.33301 5.61946 1.33301 7.16664C1.33301 8.71381 1.94762 10.1976 3.04164 11.2916C4.13566 12.3857 5.61946 13.0003 7.16664 13.0003C8.71381 13.0003 10.1976 12.3857 11.2916 11.2916Z"
                        stroke="#4B4B4B"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="1.15972"
                      />
                    </svg>
                    <input
                      class="c9"
                      data-testid="SearchBarInput"
                      placeholder="Search buildings..."
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="c10"
                />
              </div>
            </div>
            <div
              aria-labelledby="displaylist"
              class="c11"
              role="list"
            >
              <div
                class="c12"
                role="listitem"
              >
                <img
                  alt="building"
                  class="c13"
                  src="/src/assets/img/BuildingImg.png"
                />
                <div
                  class="c14"
                >
                  <h3
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                  >
                    Business Garden
                  </h3>
                  <p
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                  >
                    Opole, Poland
                  </p>
                  <div
                    class="c15"
                  >
                    <a
                      class=""
                      href="/buildings/4"
                    >
                      <button
                        class="c16"
                      >
                        Open details
                      </button>
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="c12"
                role="listitem"
              >
                <img
                  alt="building"
                  class="c13"
                  src="/src/assets/img/BuildingImg.png"
                />
                <div
                  class="c14"
                >
                  <h3
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                  >
                    Business Garden Poznań
                  </h3>
                  <p
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                  >
                    Poznań, Poland
                  </p>
                  <div
                    class="c15"
                  >
                    <a
                      class=""
                      href="/buildings/2"
                    >
                      <button
                        class="c16"
                      >
                        Open details
                      </button>
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="c12"
                role="listitem"
              >
                <img
                  alt="building"
                  class="c13"
                  src="/src/assets/img/BuildingImg.png"
                />
                <div
                  class="c14"
                >
                  <h3
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                  >
                    Business Garden Warszawa
                  </h3>
                  <p
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                  >
                    Warszawa, Poland
                  </p>
                  <div
                    class="c15"
                  >
                    <a
                      class=""
                      href="/buildings/5"
                    >
                      <button
                        class="c16"
                      >
                        Open details
                      </button>
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="c12"
                role="listitem"
              >
                <img
                  alt="building"
                  class="c13"
                  src="/src/assets/img/BuildingImg.png"
                />
                <div
                  class="c14"
                >
                  <h3
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                  >
                    Business Garden Wrocław
                  </h3>
                  <p
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                  >
                    Wrocław, Poland
                  </p>
                  <div
                    class="c15"
                  >
                    <a
                      class=""
                      href="/buildings/1"
                    >
                      <button
                        class="c16"
                      >
                        Open details
                      </button>
                    </a>
                  </div>
                </div>
              </div>
              <div
                class="c12"
                role="listitem"
              >
                <img
                  alt="building"
                  class="c13"
                  src="/src/assets/img/BuildingImg.png"
                />
                <div
                  class="c14"
                >
                  <h3
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                  >
                    Business Park
                  </h3>
                  <p
                    style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                  >
                    Kraków, Poland
                  </p>
                  <div
                    class="c15"
                  >
                    <a
                      class=""
                      href="/buildings/3"
                    >
                      <button
                        class="c16"
                      >
                        Open details
                      </button>
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="location-display"
      >
        /
      </div>
    </div>
  </body>,
  "container": .c15 {
  margin-top: 1rem;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #fff;
  border-radius: 0.5rem;
  color: #1B1A1A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c6:hover {
  background-color: #fff;
  border: 1px solid #fff;
  color: #6573FF;
}

.c16 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4553DD;
  border: 1px solid #4553DD;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 10px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.5625rem 1rem;
  white-space: nowrap;
  width: 100%;
  font-family: Source Sans Pro,sans-serif;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c16:hover {
  background-color: #6573FF;
  border: 1px solid #6573FF;
}

.c12 {
  background-color: #fff;
  border-radius: 8px;
  box-shadow: 0px 5px 8px #D9D9D920;
  color: #4B4B4B;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c13 {
  height: 11rem;
  object-fit: cover;
  padding: 0.5rem;
  width: 100%;
}

.c14 {
  padding: 0.5rem 1.031rem 0.938rem;
}

.c11 {
  display: grid;
  gap: 2rem;
  grid-template-columns: repeat(auto-fit,minmax(20rem,1fr));
}

.c0 {
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1rem;
  position: relative;
  width: 100%;
  z-index: 1;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  width: 100%;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  border: 1px solid #fff;
  border-bottom: 1px solid transparent;
  border-radius: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0.375rem 0rem 0.375rem 0.875rem;
  width: 17.5rem;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  gap: 0.5625rem;
  height: 1.75rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c9 {
  font-family: Source Sans Pro,sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.4;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  width: 100%;
}

.c10 {
  position: relative;
}

.c5 {
  position: relative;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 1.5rem 0 1rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

<div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <h2
          style="font-family: Poppins, sans-serif; font-weight: 700; font-size: 1.5rem; line-height: 1.4; color: rgb(27, 26, 26);"
        >
          Buildings
        </h2>
      </div>
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
          >
            <div
              class="c5"
            >
              <button
                class="c6"
              >
                <svg
                  data-testid="arrowDown-icon"
                  fill="none"
                  height="12"
                  viewBox="0 0 12 7"
                  width="12"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <mask
                    id="mask-arrowDown"
                  />
                  <path
                    d="M10.6667 1.33325L6.08333 5.49992L1.5 1.33325"
                    stroke="#1B1A1A"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.67"
                  />
                </svg>
                Sort Buildings
              </button>
            </div>
            <div>
              <div
                class="c7"
              >
                <div
                  class="c8"
                >
                  <svg
                    data-testid="search-icon"
                    fill="none"
                    height="20"
                    viewBox="0 0 16 16"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <mask
                      id="mask-search"
                    />
                    <path
                      d="M14.6666 14.6666L11.2916 11.2916M11.2916 11.2916C11.8333 10.7499 12.263 10.1068 12.5562 9.39907C12.8494 8.6913 13.0003 7.93272 13.0003 7.16664C13.0003 6.40055 12.8494 5.64197 12.5562 4.9342C12.263 4.22644 11.8333 3.58334 11.2916 3.04164C10.7499 2.49994 10.1068 2.07023 9.39907 1.77707C8.6913 1.4839 7.93272 1.33301 7.16664 1.33301C6.40055 1.33301 5.64197 1.4839 4.9342 1.77707C4.22644 2.07023 3.58334 2.49994 3.04164 3.04164C1.94762 4.13566 1.33301 5.61946 1.33301 7.16664C1.33301 8.71381 1.94762 10.1976 3.04164 11.2916C4.13566 12.3857 5.61946 13.0003 7.16664 13.0003C8.71381 13.0003 10.1976 12.3857 11.2916 11.2916Z"
                      stroke="#4B4B4B"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="1.15972"
                    />
                  </svg>
                  <input
                    class="c9"
                    data-testid="SearchBarInput"
                    placeholder="Search buildings..."
                    value=""
                  />
                </div>
              </div>
              <div
                class="c10"
              />
            </div>
          </div>
          <div
            aria-labelledby="displaylist"
            class="c11"
            role="list"
          >
            <div
              class="c12"
              role="listitem"
            >
              <img
                alt="building"
                class="c13"
                src="/src/assets/img/BuildingImg.png"
              />
              <div
                class="c14"
              >
                <h3
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                >
                  Business Garden
                </h3>
                <p
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                >
                  Opole, Poland
                </p>
                <div
                  class="c15"
                >
                  <a
                    class=""
                    href="/buildings/4"
                  >
                    <button
                      class="c16"
                    >
                      Open details
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="c12"
              role="listitem"
            >
              <img
                alt="building"
                class="c13"
                src="/src/assets/img/BuildingImg.png"
              />
              <div
                class="c14"
              >
                <h3
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                >
                  Business Garden Poznań
                </h3>
                <p
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                >
                  Poznań, Poland
                </p>
                <div
                  class="c15"
                >
                  <a
                    class=""
                    href="/buildings/2"
                  >
                    <button
                      class="c16"
                    >
                      Open details
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="c12"
              role="listitem"
            >
              <img
                alt="building"
                class="c13"
                src="/src/assets/img/BuildingImg.png"
              />
              <div
                class="c14"
              >
                <h3
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                >
                  Business Garden Warszawa
                </h3>
                <p
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                >
                  Warszawa, Poland
                </p>
                <div
                  class="c15"
                >
                  <a
                    class=""
                    href="/buildings/5"
                  >
                    <button
                      class="c16"
                    >
                      Open details
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="c12"
              role="listitem"
            >
              <img
                alt="building"
                class="c13"
                src="/src/assets/img/BuildingImg.png"
              />
              <div
                class="c14"
              >
                <h3
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                >
                  Business Garden Wrocław
                </h3>
                <p
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                >
                  Wrocław, Poland
                </p>
                <div
                  class="c15"
                >
                  <a
                    class=""
                    href="/buildings/1"
                  >
                    <button
                      class="c16"
                    >
                      Open details
                    </button>
                  </a>
                </div>
              </div>
            </div>
            <div
              class="c12"
              role="listitem"
            >
              <img
                alt="building"
                class="c13"
                src="/src/assets/img/BuildingImg.png"
              />
              <div
                class="c14"
              >
                <h3
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 700; font-size: 1.25rem; line-height: 1.4; letter-spacing: 1px; color: rgb(75, 75, 75);"
                >
                  Business Park
                </h3>
                <p
                  style="font-family: Source Sans Pro, sans-serif; font-weight: 400; font-size: 1rem; line-height: 1.6; letter-spacing: 0; color: rgb(75, 75, 75);"
                >
                  Kraków, Poland
                </p>
                <div
                  class="c15"
                >
                  <a
                    class=""
                    href="/buildings/3"
                  >
                    <button
                      class="c16"
                    >
                      Open details
                    </button>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="location-display"
    >
      /
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
